server/pipeline/create.go

```diff
@@ -85,10 +85,12 @@ func Create(ctx context.Context, _store store.Store, repo *model.Repo, pipeline
 
 	if configFetchErr != nil {
 		log.Debug().Str("repo", repo.FullName).Err(configFetchErr).Msgf("cannot find config '%s' in '%s' with user: '%s'", repo.Config, pipeline.Ref, repoUser.Login)
+		err := &ErrConfigNotFound{Msg: fmt.Sprintf("pipeline definition not found in %s", repo.FullName)}
 		pipeline.Started = time.Now().Unix()
 		pipeline.Finished = pipeline.Started
 		pipeline.Status = model.StatusError
-		pipeline.Error = fmt.Sprintf("pipeline definition not found in %s", repo.FullName)
+		pipeline.Error = err.Msg
+		return nil, err
```

Comment:
Why not just add?

```suggestion
		if pipeline.Branch != repo.Branch {
			return nil, err
		}
```

Reply:
Why not just let the user control it? Why should we assume that the default branch requires a CI config for every user or setup? We can add options to disable/enable this feature later right now irgnoring it and logging an error in the backend is good enough IMO.

Reply:
Not for my use case, this will increase our support team work on finding out why pipelines are not working for new projects because even with current error that config was not found there were some developers that asked for help. Now it will be even harder to find that as developers won't see anything and it will just not work silently

Reply:
I understand your point but I still disagree. However not my decision, Ill leave this PR to someone else to pick up or make a decision.

Reply:
But from what you wrote previously this would work also for you so this would be good middle ground that would be good and not too breaking change and later on as you said this could be improved for other cases if needed